docs/markdown/Python/python/python-lockfiles.md

```diff
@@ -0,0 +1,271 @@
+---
+title: "Lockfiles"
+slug: "python-lockfiles"
+excerpt: "Securely locking down your third-party dependencies."
+hidden: false
+createdAt: "2023-03-11T01:50:46.369Z"
+updatedAt: "2023-03-11T01:50:46.369Z"
+---
+Third-party dependencies are typically specified via a range of allowed versions, known as "requirements", in a file such as requirements.txt or pyproject.toml. A dependency resolution tool like Pip or Poetry then takes these initial requirements and attempts to find and download a consistent set of transitive dependencies that are mutually compatible with each other and with the target Python interpreter version.
+
+When used naively, this dependency resolution process is unstable: if you run a resolve, and then some time later run another resolve on the same inputs, you may end up with a different resulting set of dependencies. This is because new versions of direct or transitive dependencies may have been published (or, in rare cases, yanked) between the two runs.
```

Comment:
Poetry always locks IIUC even without you telling it to do so; so lumping it in with Pip above is a bit unfair / inaccurate.